### ISSUE TITLE
Only a few contacts and no photos in Maps

### ISSUE THREAD

**Issue by Martin511** [nextcloud/maps#635]
Hello.
I see only 8 of 270 contacts in maps and no photos.
Is there something special to edit to show all contacts and photos in maps?
I use Nextcloud 21.0.3 and maps 0.1.9. I see nor error messages.

**Comment by bartart3d** [comment]
Nextcloud 21.0.2 and maps 0.1.9 here,  I just ran the occ command to scan the photo's and no photo's appear on the map here either. Even though the terminal showed all the files the command could find, none of them show up on the map.
My pictures are not stored in the default photo's map, since I'm running a raspberry and think it is safer to keep files stored on an external drive instead of on a flash card... Could this be the issue?

**Comment by Martin511** [comment]
@bartart3d 
My installation is also running on a Raspberry with the Photos on external drive.

**Comment by nunesgh** [comment]
Same problem here on Nextcloud 21.0.3 and Maps 0.1.9.
Only 2 of my contacts appear on Maps, out of several with the Address property set.

Also reported on https://github.com/nextcloud/maps/issues/511.

**Comment by JMLatGitHub** [comment]
Hi all, any kind of feedback on that issue would be greatly appreciated. Or even a hint for a workaround or troubleshooting. Or is the development of the Nextcloud maps app maybe totally abandoned?

Kind regards
JML

**Comment by kinslayer1982** [comment]
Manually check the address on the map. I noticed that street names are wrong in the app. It adds other things like city district to the streets name. Checking the address in openstreetmap wil show the correct street name and additional tags, some of which will be appended to the streets name by the app.

**Comment by Martin511** [comment]
Hi kinslayer1982,
it is not about that the contacts appear wron in the maps app.
They don´t appear alt all.
I think a function like a re-scan contacts occ command is worth a try.

**Comment by kinslayer1982** [comment]
> 
> 
> Hi kinslayer1982, it is not about that the contacts appear wron in the maps app. They don´t appear alt all. I think a function like a re-scan contacts occ command is worth a try.

But that's exactly what I'm talking about. As the map app has the street name wrong it will not display your contact at all, because the (correct) address that is saved in your contact details does not exist for the app.

Try it: Manually got to the address of your contact in the app, click on the building and add the address to your contact via the map app. It will instantly display your contact.

**Comment by Spartachetto** [comment]
My advice is to check two things:
1. Does the address exist in OpenStreetMap (https://www.openstreetmap.org) ? If it doesn't exist you could try to map it by looking at https://www.openstreetmap.org/help 
2. Does Nominatim (http://nominatim.org/ ) find it? As @Martin511 says, Nominatim could be a bit picky and you can try to write the address in a way it likes...

**Comment by kinslayer1982** [comment]
> 
> 
> My advice is to check two things:
> 
>     1. Does the address exist in OpenStreetMap (https://www.openstreetmap.org) ? If it doesn't exist you could try to map it by looking at https://www.openstreetmap.org/help
> 
>     2. Does Nominatim (http://nominatim.org/ ) find it? As @Martin511 says, Nominatim could be a bit picky and you can try to write the address in a way it likes...

Yes, both openstreetmap and nominatim can find the address just fine, only the map app doesn't get the address right.

**Comment by Martin511** [comment]
> Manually got to the address of your contact in the app, click on the building and add the address to your contact via the map app. It will instantly display your contact.

Ok, if I do this, it works. BUT: The address then is formatted in english layout (For example: german street with number: "Hauptstraße 10" vs. english street with numer: "10 Hauptstraße").
I live in Germany and I really want to keep the german address layout.
So is it all about the german layout that keep contacts from being displayed?

**Comment by kinslayer1982** [comment]
> 
> 
> > Manually got to the address of your contact in the app, click on the building and add the address to your contact via the map app. It will instantly display your contact.
> 
> Ok, if I do this, it works. BUT: The address then is formatted in english layout (For example: german street with number: "Hauptstraße 10" vs. english street with numer: "10 Hauptstraße"). I live in Germany and I really want to keep the german address layout. So is it all about the german layout that keep contacts from being displayed?

I'm from Germany too.
The more I try to figure it out, the more fucked up this whole system seems.
Let me give some examples:
- "Fasanenstraße 33" will show up, changing it to "33 Fasanenstraße" will not. Changing it back will work again.
- The street from the contact is "Sandstraße 37a", it doesn't show. Navigating to the address and clicking on it, then adding to the contact will result in "37a Sandstraße Opladen Stadtbezirk II" being saved as street. Changing it to "Sandstraße 37a Opladen Stadtbezirk II" does not work. Changing it back to "37a Sandstraße Opladen Stadtbezirk II" does not work. Only works by adding the address via the map app.
- Street is "Wäschlacker Weg 28". Doesn't work. Navigating to the address and adding it will result in street "28 Wäschlacker Weg Lierenfeld Stadtbezirk 8". Putting the number after the street name doesn't work. Changing it back doesn't work, just like the address mentioned above.
- There's an address that works by putting the postal code and city together with the street name and number like "<street><number>,<postal code><city>" but putting street and number as street, postal code as postal code and city as city will not work.

**Comment by charlescurley** [comment]
A bit of detective work.... 1230 East _First_ Street, Casper Wyoming, does not work. However, 1230 East _1st_ Street does. One problem with this is that the latter is the preferred form in Casper, and how the contact at that address provided the address.

Maybe some attempt at alternates is in order? I realize this would be a huge, trial and error, task. But something to think about.

**Comment by tintou** [comment]
I believe that this fixes the Contacts part https://github.com/nextcloud/maps/pull/1465
